Mathlib/Topology/Instances/CantorSet.lean

```diff
@@ -0,0 +1,92 @@
+/-
+Copyright (c) 2024 Jana Göken. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Artur Szafarczyk, Suraj Krishna M S, Jean-Baptiste Stiegler, Isabelle Dubois,
+Tomáš Jakl, Lorenzo Zanichelli, Alina Yan, Emilie Uthaiwat, Jana Göken,
+Filippo A. E. Nuccio
+-/
+import Mathlib.Topology.Metrizable.Basic
+import Mathlib.Topology.Algebra.GroupWithZero
+import Mathlib.Topology.Instances.Real
+
+/-!
+# Ternary Cantor Set
+
+This file defines the Cantor ternary set and proves a few properties.
+
+## Main Definitions
+
+* `preCantorSet n`: The order `n` pre-Cantor set, defined inductively as the union of the images
+  under the functions `(· / 3)` and `(2 * · / 3)`, with `preCantorSet 0 := Set.Icc 0 1`, i.e.
+  `preCantorSet 0` is the unit interval [0,1].
+* `cantorSet`: The ternary Cantor set, defined as the intersection of all pre-Cantor sets.
+-/
+
+/-- The definition of preCantorSet, as explained above. -/
```

Comment:
Could you copy the doc-string from above here, otherwise the hover-over is annoying.

Reply:
You mean writing the definition of ``preCantorSet`` here again, i.e. having the same thing twice?

Reply:
Yes, this (modulo linebreak):
```suggestion
/-- The order `n` pre-Cantor set, defined inductively as the union of the images under the functions `(· / 3)` and `(2 * · / 3)`, with `preCantorSet 0 := Set.Icc 0 1`, i.e. `preCantorSet 0` is the unit interval [0,1]. -/
```

Reply:
@jaklt yes, it needs to be duplicated. The first one is for the website documentation so that someone reading only the top can comprehend what's going on. 

The do string on the `def` itself also goes on the website (so the info is duplicates there), but crucially, this docstring is also displayed when you hover over `preCantorSet` in VS Code, even in a different file. This is why it's important.